### PR TITLE
refactor(ekf_localizer): rework parameters

### PR DESCRIPTION
**Pull request by Motsu-san**
## Description
This pull request must be merged with [the autoware_launch pull request](https://github.com/autowarefoundation/autoware_launch/pull/847).
Please see both pull request.

Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371.

- Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.
  - already completed.
- Create the yaml configs in this package and the autoware_launch repository.
  - already completed.
- Create the schema.
  - This PR.
- Update `Parameters` chapter in the readme file.
  - This PR.

### confirm parameters
1. run autoware & replay rosbag that can start localization(Autoware sample rosbag is recommended).
2. command the following
```
ros2 param dump /localization/pose_twist_fusion_filter/ekf_localizer 
```
expected result
```
/localization/pose_twist_fusion_filter/ekf_localizer:
  ros__parameters:
    diagnostics:
      pose_no_update_count_threshold_error: 100
      pose_no_update_count_threshold_warn: 50
      twist_no_update_count_threshold_error: 100
      twist_no_update_count_threshold_warn: 50
    misc:
      threshold_observable_velocity_mps: 0.0
    node:
      enable_yaw_bias_estimation: true
      extend_state_step: 50
      predict_frequency: 50.0
      publish_tf: true
      show_debug_info: false
      tf_rate: 50.0
    pose_frame_id: map
    pose_measurement:
      pose_additional_delay: 0.0
      pose_gate_dist: 10000.0
      pose_smoothing_steps: 5
    process_noise:
      proc_stddev_vx_c: 10.0
      proc_stddev_wz_c: 5.0
      proc_stddev_yaw_c: 0.005
    qos_overrides:
      /clock:
        subscription:
          depth: 1
          durability: volatile
          history: keep_last
          reliability: best_effort
      /parameter_events:
        publisher:
          depth: 1000
          durability: volatile
          history: keep_last
          reliability: reliable
      /tf:
        publisher:
          depth: 100
          durability: volatile
          history: keep_last
          reliability: reliable
    simple_1d_filter_parameters:
      pitch_filter_proc_dev: 0.01
      roll_filter_proc_dev: 0.01
      z_filter_proc_dev: 1.0
    twist_measurement:
      twist_additional_delay: 0.0
      twist_gate_dist: 10000.0
      twist_smoothing_steps: 2
    use_sim_time: true

```

### Confirm json schema
1. Confirm if check-jsonschema is installed.
```
pip3 install -U check-jsonschema
```
2. Command the following
```
cd [Your Autoware.universe dir]/localization/ekf_localizer
find -wholename '*/schema/*.schema.json' -printf '%p: ' -execdir bash -c 'check-jsonschema -v --schemafile "$1" ../config/"${1:2:-12}"*.param.yaml' bash '{}' +
```
result
```
localization/ekf_localizer$ find -wholename '*/schema/*.schema.json' -printf '%p: ' -execdir bash -c 'check-jsonschema -v --schemafile "$1" ../config/"${1:2:-12}"*.param.yaml' bash '{}' +
./schema/ekf_localizer.schema.json: ok -- validation done
The following files were checked:
  ../config/ekf_localizer.param.yaml
```
### Confirm README display
1. Ready to use ```mkdocs serve``` with [this instruction](https://autowarefoundation.github.io/autoware-documentation/main/contributing/documentation-guidelines/#2-running-an-mkdocs-server-in-your-local-environment).
2. Command the following
```
cd [Your Autoware.universe dir]
mkdocs serve
```

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp [hunk 1]
```diff
@@ -24,36 +24,41 @@ class HyperParameters
...
     ekf_dt(1.0 / std::max(ekf_rate, 0.1)),
-    tf_rate_(node->declare_parameter("tf_rate", 10.0)),
-    publish_tf_(node->declare_parameter("publish_tf", true)),
-    enable_yaw_bias_estimation(node->declare_parameter("enable_yaw_bias_estimation", true)),
-    extend_state_step(node->declare_parameter("extend_state_step", 50)),
+    tf_rate_(node->declare_parameter<double>("node.tf_rate")),
+    publish_tf_(node->declare_parameter<bool>("node.publish_tf")),
+    enable_yaw_bias_estimation(node->declare_parameter<bool>("node.enable_yaw_bias_estimation")),
+    extend_state_step(node->declare_parameter<int>("node.extend_state_step")),
     pose_frame_id(node->declare_parameter("pose_frame_id", std::string("map"))),
```
Comment (KYabuuchi) [inline localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp:34]: Please delete the default value of "pose_frame_id" as well, and change it to be specified in param.yaml.
In addition, remove the "pose_frame_id" from the [`ekf_localizer.launch.xml`](https://github.com/autowarefoundation/autoware.universe/blob/bd4b5ca2fc48c972fabb5fff68bdcbf5b1aeece8/localization/ekf_localizer/launch/ekf_localizer.launch.xml#L28) :pray:

**Review by KYabuuchi — COMMENTED** [review]
The `pose_frame_id` parameter has a default value so please modify it. :pray:
Except for that, there is no problem

File: localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp (hunk 1 above)
Comment (Motsu-san) [reply localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp]: @KYabuuchi Thank you for your review. I modified it just now. Please check it.

Comment (KYabuuchi) [reply localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp]: @Motsu-san  Thanks. But you need to modify `misc.sub_schema.json` too :pray:

Comment (Motsu-san) [reply localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp]: @KYabuuchi Yeah, I'm aware of it. Before this pull request, there was not "pose_frame_id" in the Misc table. That's why I didn't. I would add "pose_frame_id" description to README if allowed. So, I would like to know what it means and how to set if needed. Could you?
image

Comment (KYabuuchi) [reply localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp]: @Motsu-san  Please add it to the Misc table. "pose_frame_id" means `Parent frame_id of EKF output pose`.

Comment (Motsu-san) [reply localization/ekf_localizer/include/ekf_localizer/hyper_parameters.hpp]: @KYabuuchi OK. I'll do that. I understood that it is needed to be changed when just using other frame_id of output pose than "map". Thanks.

**Review by KYabuuchi — APPROVED** [review]
LGTM :ok_person:

**Pull request merged**